harness/determined/pytorch/deepspeed/_deepspeed_trial.py

```diff
@@ -675,29 +1071,53 @@ def _save(self, path: pathlib.Path) -> None:
 
         if torch.cuda.device_count():
             rng_state["gpu_rng_state"] = torch.cuda.get_rng_state(
-                self.context.distributed.get_local_rank()
+                self.context.distributed.local_rank
             )
-        checkpoint = {"rng_state": rng_state}
 
         # PyTorch uses optimizer objects that take the model parameters to
         # optimize on construction, so we store and reload the `state_dict()`
         # of the model and optimizer explicitly (instead of dumping the entire
         # objects) to avoid breaking the connection between the model and the
         # optimizer.
-        checkpoint["callbacks"] = {
-            name: callback.state_dict() for name, callback in self.callbacks.items()
+        checkpoint = {
+            "callbacks": {name: callback.state_dict() for name, callback in self.callbacks.items()},
+            "rng_state": rng_state,
         }
 
         for callback in self.callbacks.values():
             callback.on_checkpoint_save_start(checkpoint)
-
         ckpt_name = f"det_state_dict_rank{self.context.distributed.rank}.pth"
         torch.save(checkpoint, str(path.joinpath(ckpt_name)))
 
+        assert self.state
+        with path.joinpath("trial_state.pkl").open("wb") as f:
+            pickle.dump(vars(self.state), f)
+
         # We allow users to override save behavior if needed, but we default to using
         # the save method provided by DeepSpeed.
         self.trial.save(self.context, path)
 
+        trial_cls = type(self.trial)
+        with open(path.joinpath("load_data.json"), "w") as f2:
+            try:
+                exp_conf = self.context.get_experiment_config()  # type: Optional[Dict[str, Any]]
+                hparams = self.context.get_hparams()  # type: Optional[Dict[str, Any]]
+            except ValueError:
+                exp_conf = None
+                hparams = None
+
+            load_data = {
+                "trial_type": "DeepSpeedTrial",
+                "experiment_config": exp_conf,
+                "hparams": hparams,
+                "trial_cls_spec": f"{trial_cls.__module__}:{trial_cls.__qualname__}",
```

Comment:
so, this exists because we have `load_trial_from_checkpoint_path` in `pytorch/_load.py` which basically lets users load a checkpoint outside of a running trial, which had to be modified with PyTorch Trainer since users can instantiate their own trial classes now (with arbitrary arguments) and there's no easy way to load a checkpoint. 

i don't think we ever supported this sort of external checkpoint loading option for deepspeed, so i don't think it's worth doing now, but noting this as a potential future improvement.

but it means we don't actually need this in the checkpoint, or all the `is_trainer` stuff